Sources/MessagingPush/MessagingPushConfigOptions.swift

```diff
@@ -1,21 +1,33 @@
-public struct MessagingPushConfigOptions {
-    // Used to create new instance of MessagingPushConfigOptions when the MessagingPushAPN module is configured.
-    // Each property of the MessagingPushConfigOptions object can be modified by the user.
-    public enum Factory {
-        public static func create() -> MessagingPushConfigOptions {
-            MessagingPushConfigOptions(
-                autoFetchDeviceToken: true
-            )
-        }
-    }
+import CioInternalCommon
 
-    public enum Keys: String { // Constants used to map each of the options in MessagingPushConfigOptions
-        case autoFetchDeviceToken
+public struct MessagingPushConfigOptions {
+    public init() {
+        self.autoFetchDeviceToken = true
+        self.showPushAppInForeground = true
     }
 
     /**
      Enable automatic fetching of device token by the SDK without the need to write custom code by the customer.
      On fetching the token, SDK will auto-register the device. This value is `true` by default.
      */
     public var autoFetchDeviceToken: Bool
+
+    /**
+     Display push notifications sent by Customer.io while app is in foreground. This value is `true` by default.
+     */
+    public var showPushAppInForeground: Bool
+}
+
+// Add MessagingPush config options to the DIGraph like we do for SdkConfig.
+// Allows dependencies to easily access module configuration via dependency injection
+// in constructor.
+extension DIGraph {
+    var messagingPushConfigOptions: MessagingPushConfigOptions {
```

Comment:
In Android, we have an option like `moduleConfigOptions` which is scoped to module. So classes within in-app module can use the same name for its config, while push can access its config with same name at same time. Can we apply the same here?

Reply:
Unfortunately, the naming of the variable must match the name of the protocol, in camelCase. That's the naming convention that the DI graph code generator expects. 

However, this name is only for *including* the dependency in the graph. When dependencies need to use this dependency, the name can be whatever you want. 

Example: 
```swift
class Foo {
  let moduleConfigOptions: MessagingPushConfigOptions

  init(whatever: MessagingPushConfigOptions} {
    self.moduleConfigOptions = whatever
  }
```

So, for the majority of the code base, yes, you can use `moduleConfigOptions` naming.

Reply:
Makes sense. Thanks for the clarification 👍🏻